modules/appeals_api/app/models/concerns/appeals_api/contestable_issue.rb

```diff
@@ -0,0 +1,63 @@
+# frozen_string_literal: true
+
+class AppealsApi::ContestableIssue
+  def initialize(issue)
+    @issue = issue
+  end
+
+  def decision_date
+    return unless decision_date_string
+
+    parse_date(decision_date_string)
+  end
+
+  def decision_date_string
+    issue.dig('attributes', 'decisionDate')
+  end
+
+  def soc_date
+    return unless soc_date_string
+
+    parse_date(soc_date_string)
+  end
+
+  def soc_date_string
+    issue.dig('attributes', 'socDate')
+  end
+
+  def text
+    issue.dig('attributes', 'issue')
+  end
+
+  delegate :[], to: :issue
+
+  def text_exists?
+    text.present?
+  end
+
+  def soc_date_past?
```

Comment:
While we're at it, let's combine the various soc/text/decision_date methods together, since this is the first commit that introduces them.

Reply:
I don't really understand how these are supposed to combine, or the reasoning you're presenting for it. Could you elaborate a bit more?

Reply:
Ah, sorry, I can see how "combine" could mean something more than I intended. I just meant move the methods around so that all the SOC methods are grouped together, etc.

Reply:
Ah, got it.